File: backend/src/main/java/info/gamewise/lor/videos/domain/VideoChampion.java
package info.gamewise.lor.videos.domain;

import info.gamewise.lor.videos.domain.json.Champion;

public final class VideoChampion {
    private final String code;
    private final String name;
    private final String urlImgName;

    public VideoChampion(Champion champion) {
        this.code = champion.code();
        this.name = champion.name();
        this.urlImgName = buildUrlImgName(champion);
    }

    public String getCode() {
        return code;
    }

    public String getName() {
        return name;
    }

    public String getUrlImgName() {
        return urlImgName;
    }

    @Override
    public String toString() {
        return "VideoChampion{" +
                "code='" + code + '\'' +
                ", name='" + name + '\'' +
                ", urlImgName='" + urlImgName + '\'' +
                '}';
    }

    private String buildUrlImgName(Champion champion) {
        return champion
                .name()
                .replace(" ", "");
    }
}

File: backend/src/main/kotlin/info/gamewise/lor/videos/domain/VideoChampion.kt
package info.gamewise.lor.videos.domain

import info.gamewise.lor.videos.domain.json.Champion

class VideoChampion(champion: Champion) {

    private val championsWithIdAsUrl = arrayOf("GWEN", "NORRA")

    val code: String
    val name: String
    val urlImg: String

    init {
        this.code = champion.code
        this.name = champion.name
        this.urlImg = buildUrlImgName(champion)
    }

    override fun toString(): String {
        return "VideoChampion{" +
                "code='" + code + '\'' +
                ", name='" + name + '\'' +
                ", urlImage='" + urlImg + '\'' +
                '}'
    }

    private fun buildUrlImgName(champion: Champion): String {
        if (championsWithIdAsUrl.contains(champion.code)) {
            return "https://cdn-lor.mobalytics.gg/production/images/round-cards/${champion.id}.png"
        }

        val cardName = when (champion.code) {
            "KAI_SA" -> "Kaisa"
            else -> champion.name.replace(" ", "")
        }

        return "https://cdn-lor.mobalytics.gg/production/images/champions/$cardName.webp"
    }
}